go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -239,6 +239,10 @@ def handle_subprocess_error(stderr) # rubocop:disable Metrics/AbcSize
             raise Dependabot::DependencyFileNotResolvable, error_message
           end
 
+          if (matches = stderr.match(/Authentication failed for '(?<url>.+)'/))
```

Comment:
I moved this up to fix this test which was providing invalid credentials: https://github.com/dependabot/dependabot-core/blob/main/go_modules/spec/dependabot/go_modules/file_updater_spec.rb#L137

I think it also makes sense to prioritize reporting that your credentials are invalid since that could be the root cause that the git deps aren't reachable.